main.go

```diff
@@ -66,8 +66,40 @@ func main() {
 	outputFile := flag.String("o", "", "optional output to file. Default is stdout")
 	var skipDependencies flagsSet = map[string]bool{}
 	flag.Var(&skipDependencies, "skip-dependency", "Dependency to skip generating wag clients. Can be added mulitple times e.g. -skip-dependency a -skip-dependency b")
+	overrideDependenciesString := flag.String("d", "", "Dependency name to override. You can provide multiple dependencies in the format dep1:replacementDep1,dep2:replacementDep2,...")
 	flag.Parse()
 
+	// parsing through the list of overrides to make an original:new string map
+	var overrideDependenciesList []string
+	if len(*overrideDependenciesString) > 3 {
+		overrideDependenciesList = strings.Split(*overrideDependenciesString, ",")
+	}
+	overrideDependenciesMap := make(map[string]string)
+
+	for _, s := range overrideDependenciesList {
+		tempArr := strings.Split(s, ":")
+
+		if len(tempArr) != 2 || tempArr[1] == "" {
+			log.Fatal("usage: invalid formatting for the -d flag")
+		}
+
+		flag := 0
+		for _, d := range t.Dependencies {
+			if d == tempArr[0] {
+				flag = 1
+			}
+		}
+
+		if flag == 0 { // if a dependency provided along with the override flag is not in the dependencies on the yml
+			log.Fatal("that is not a dependency specified in the yml provided")
+		}
+
+		original := tempArr[0]
+		new := tempArr[1]
+
+		overrideDependenciesMap[original] = new
```

Comment:
Can just write `overrideDependenciesMap[tempArr[0]] = tempArr[1]`

Reply:
fixing that... i initially did that for readability